src/Hng.Web/Controllers/AuthController.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Hng.Application.Features.UserManagement.Dtos;
+using Hng.Application.Features.UserManagement.Commands;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Hng.Web.Controllers
+{
+    [ApiController]
+    [Route("api/v1/auth")]
+    public class AuthController : ControllerBase
+    {
+        private readonly IMediator _mediator;
+
+        public AuthController(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+
+        [HttpPost("register")]
+        [ProducesResponseType(typeof(SignUpResponse), StatusCodes.Status201Created)]
+        [ProducesResponseType(typeof(SignUpResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(SignUpResponse), StatusCodes.Status422UnprocessableEntity)]
+        [ProducesResponseType(typeof(SignUpResponse), StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult<SignUpResponse>> UserSignUp([FromBody] UserSignUpDto body)
+        {
+            if (!ModelState.IsValid)
```

Comment:
No need for this. The Data annotation attribute takes care of the validation

Reply:
/register  moved to AuthenticationController.cs